### PR TITLE
Tensorflow upgrade documentation

### PR DESCRIPTION
**Pull request by c0derlint**
GCI [Task](https://codein.withgoogle.com/dashboard/task-instances/6595974689456128/): Upgrade TensorFlow to a higher version

This PR simply adds the markdown file explaining the tensorflow upgrade.

#429 - Updated Requirements.

### CODE REVIEW

**Comment by Ram81** [comment]
@thatbrguy we would need both environments to be up to date. It would be a good idea to update docker and venv and try all model functionalities.

File: tensorflow_upgrade.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+## Tensorflow upgrade
+
+Current Version: 1.4.1
+
+Versions Tested:
+ - 1.5.0
+ - 1.6.0
+ - 1.7.0
+ - 1.8.0 Upgrading to 1.8.0 breaks no functionality and produces no error when running with Travis CI. Importing data models also do not seem to produce any problems.
+ - 1.9.0 & 1.10.1 These versions gave me an _ImportError: dlopen: cannot load any more object with static TLS_ Looking online, this seemed to be a bug associated with these versions. On upgrading to 1.12.0, this issue no longer persisted. (Travis CI log: [1.9.0](https://travis-ci.com/c0derlint/Fabrik/builds/90459224), [1.10.1](https://travis-ci.com/c0derlint/Fabrik/builds/90600645))
```
Comment (Ram81) [inline tensorflow_upgrade.md]: Can you rephrase the line `These versions gave me an _ImportError:` to something similar to `Errors encountered during ...`

Comment (Ram81) [reply tensorflow_upgrade.md]: Make this as two section I would say,

1. Errors encountered
2. Cause of the error

Comment (c0derlint) [reply tensorflow_upgrade.md]: Okay, give me a moment.

File: tensorflow_upgrade.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Versions Tested:
+ - **1.5.0**
+ - **1.6.0**
+ - **1.7.0**
+ - **1.8.0** 
+    - **Remarks:** Upgrading to version 1.5.0-1.8.0 breaks no functionality and produces no error when running with Travis CI. Importing data models also do not seem to produce any problems.
+ - **1.9.0** & **1.10.1** 
+    - **Errors encountered:** An `ImportError: dlopen: cannot load any more object with static TLS_` was thrown when trying to run the project.
+    - **Cause of error:** This error seems to be thrown due to a wrong order of import statements. This error seems to be a bug as none of the breaking changes from v1.8.0 to 1.9.0 ([source](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md#breaking-changes-2)) apply. On upgrading to 1.12.0, this issue no longer persisted.
```
Comment (Ram81) [inline tensorflow_upgrade.md]: Can you rephrase this `This error seems to be thrown due to a wrong order of import statements. This error seems to be a bug` as `This error seems to be thrown due to a wrong order of import statements, which seems to be a bug..`
